In [6]:
gejala_user = {
    "demam": 0.7,
    "batuk": 0.5,
    "sakit_tenggorokan": 0.6,
    "pilek": 0.8,
    "sakit_kepala": 0.5,
    "lelah": 0.6,
    "nyeri_otot": 0.4,
    "bersin": 0.7
}

pengetahuan = {
    "flu": {
        "demam": 0.8,
        "batuk": 0.7,
        "sakit_tenggorokan": 0.6,
        "pilek": 0.7,
        "sakit_kepala": 0.6,
        "lelah": 0.5,
        "nyeri_otot": 0.4,
        "bersin": 0.5
    }
}

def hitung_cf(gejala_user,pengetahuan_pakar):
  cf_total = 0
  first = True
  for gejala, cf_user in gejala_user.items():
    if gejala in pengetahuan_pakar:
      cf_pakar = pengetahuan_pakar[gejala]
      cf = cf_user * cf_pakar
      if first:
        cf_total = cf
        first = False
      else:
        cf_total = cf_total + cf * (1 - cf_total)
  return cf_total
cf_flu = hitung_cf(gejala_user,pengetahuan["flu"])
print(f"cf diagnosis flu: {cf_flu:.2f}")

cf diagnosis flu: 0.98


kode diatas adalah kode yang demam nya belum di ubah ke 0.2

In [8]:
gejala_user = {
    "demam": 0.2,
    "batuk": 0.5,
    "sakit_tenggorokan": 0.6,
    "pilek": 0.8,
    "sakit_kepala": 0.5,
    "lelah": 0.6,
    "nyeri_otot": 0.4,
    "bersin": 0.7
}

pengetahuan = {
    "flu": {
        "demam": 0.8,
        "batuk": 0.7,
        "sakit_tenggorokan": 0.6,
        "pilek": 0.7,
        "sakit_kepala": 0.6,
        "lelah": 0.5,
        "nyeri_otot": 0.4,
        "bersin": 0.5
    }
}

def hitung_cf(gejala_user,pengetahuan_pakar):
  cf_total = 0
  first = True
  for gejala, cf_user in gejala_user.items():
    if gejala in pengetahuan_pakar:
      cf_pakar = pengetahuan_pakar[gejala]
      cf = cf_user * cf_pakar
      if first:
        cf_total = cf
        first = False
      else:
        cf_total = cf_total + cf * (1 - cf_total)
  return cf_total
cf_flu = hitung_cf(gejala_user,pengetahuan["flu"])
print(f"cf diagnosis flu: {cf_flu:.2f}")

cf diagnosis flu: 0.96


kode diatas adalah kode yang demam nya telah di ubah ke 0.2
jika demam nya di ubah maka hasil cf nya adalah 0.96 karena nilai gejala pertama (dalam urutan) sangat mempengaruhi cf_total awal.
Nilai setelahnya makin teredam karena faktor (1 - cf_total_sementara).
Jadi kalau nilai awal rendah, seluruh hasil akhirnya jadi lebih rendah.